pkg/api/vm/handler.go

```diff
@@ -27,22 +36,7 @@ func (h *vmActionHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 	namespace := vars["namespace"]
 	name := vars["name"]
 
-	switch action {
-	case startVM, stopVM:
-		running := action == startVM
-		if err := h.updateVMRunningStatus(name, namespace, action, running); err != nil {
-			rw.WriteHeader(http.StatusInternalServerError)
-			rw.Write([]byte(err.Error()))
-			return
-		}
-
-		rw.WriteHeader(http.StatusNoContent)
-	case restartVM:
-		if err := h.restartVM(name, namespace); err != nil {
-			rw.WriteHeader(http.StatusInternalServerError)
-			rw.Write([]byte(err.Error()))
-		}
-	case ejectCdRom:
+	if action == ejectCdRom {
```

Comment:
Can we put this to switch as well? It is one of the cases but looks so isolated.

Reply:
Because it makes switch block big and it's a bit different, so just separate it. But I'm ok with putting it to switch

Reply:
we can put it to a separate function if that's the case. It should be better than changing the condition structure